OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -132,8 +135,24 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
 		}
 
+		void UpdateTarget()
+		{
+			if (args.GuidedTarget.IsValidFor(args.SourceActor))
+				target = args.GuidedTarget.CenterPosition;
```

Comment:
This ignores the range limit, which looks rather odd when you target a fast moving vehicle that is running away.  What was the behaviour of the original game in that case?